core/src/main/java/jenkins/model/FeatureSwitchConfiguration.java

```diff
@@ -0,0 +1,166 @@
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package jenkins.model;
+
+import hudson.Extension;
+import net.sf.json.JSONObject;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+import org.kohsuke.stapler.StaplerRequest;
+
+import java.util.*;
+
+/**
+ * Allow user to turn on/off selected experimental features. such features'
+ * behavior subject to change without notice
+ *
+ * @since TODO update version after merge
+ */
+@Restricted(NoExternalUse.class)
+@Extension
+public class FeatureSwitchConfiguration extends GlobalConfiguration {
+
+    //the feature list will be small, size less than 20
+    private List<String> disabledFeatureStr = new ArrayList<String>();
+
+    public EnumSet<Feature> getDisabledSet() {
+        return disabledSet;
+    }
+
+    /**
+     * Update the disabled feature list
+     * used by {@link #configure(StaplerRequest req, JSONObject json)}.
+     *
+     * @param disabledSet
+     */
+    public void setDisabledSet(EnumSet<Feature> disabledSet) {
+        this.disabledSet = disabledSet;
+        this.disabledFeatureStr.clear();
+        for (Feature feature : disabledSet) {
+            disabledFeatureStr.add(feature.name());
+        }
+        save();
+    }
+
+    //Allow enum to be removed in future release, avoid serialization/deserialization
+    transient EnumSet<Feature> disabledSet = EnumSet.noneOf(Feature.class);
+
+    @Override
+    public boolean configure(StaplerRequest req, JSONObject json) throws FormException {
+        req.bindJSON(this, json);
+        return true;
+    }
+
+    public FeatureSwitchConfiguration() {
+        //Load configure data from disk
+        load();
+        updateFeatureSet();
+    }
+
+    public static FeatureSwitchConfiguration get() {
+        return GlobalConfiguration.all().get(FeatureSwitchConfiguration.class);
+    }
+
+    /**
+     * Convert string to enum, removed feature(no matching enum item)
+     * will be discarded
+     */
+    private void updateFeatureSet() {
+        for (Feature feature : Feature.values()) {
+            if (disabledFeatureStr.contains(feature.name())) {
+                disabledSet.add(feature);
+            }
+        }
+    }
+
+    /**
+     * Check the feature is enabled or not
+     *
+     * @param feature
+     * @return a boolean flag indicating the feature disabled or not
+     */
+    public static boolean isDisabled(Feature feature) {
+        return get().getDisabledSet().contains(feature);
+    }
+
+    /**
+     * Always return true if feature is disabled, otherwise return the condition
+     *
+     * @param feature   the feature is enabled in the jenkins system
+     * @param condition checked condition
+     * @return true as passed, false as rejected
+     */
+    public static boolean bypass(Feature feature, boolean condition) {
```

Comment:
The function is not obvious from the name... consider `isOffOr(boolean)` and `isOnOr(boolean)` and make methods of the feature enum